tests/test_hypothesis_plots.py

```diff
@@ -18,13 +18,23 @@ def datatree(seed=31):
     tau = rng.normal(size=(3, 50, 2))
     theta = rng.normal(size=(3, 50, 2, 3))
     diverging = rng.choice([True, False], size=(3, 50), p=[0.1, 0.9])
+    obs = rng.normal(size=(2, 3))  # hierarchy, group dims respectively
+    prior_predictive = rng.normal(size=(1, 50, 2, 3))  # assuming 1 chain
+    posterior_predictive = rng.normal(size=(3, 50, 2, 3))  # all chains
 
     return from_dict(
         {
             "posterior": {"mu": mu, "theta": theta, "tau": tau},
             "sample_stats": {"diverging": diverging},
+            "observed_data": {"obs": obs},
+            "prior_predictive": {"obs": prior_predictive},
+            "posterior_predictive": {"obs": posterior_predictive},
+        },
+        dims={
+            "theta": ["chain", "draw", "hierarchy", "group"],
+            "tau": ["chain", "draw", "hierarchy"],
+            "obs": ["chain", "draw", "hierarchy", "group"],
```

Comment:
adding chain and draw is actually breaking the tests. `from_dict` assumes the `sample_dims` rcparams prepend the provided dimensions _in some of the groups_, in others, where it doesn't make sense for these dimensions to be they are not prepended. When you say `obs: chain, draw, hierarchy, group` you are breaking this behaviour, for the posterior predictive group it is not really an issue because if they are already present arviz won't prepend them again, but in observed_data they are not present at all, yet the code now says they should